### DISCUSSION TITLE
Windows: Repeated “Access is denied” errors during nnunetv2_train after moving dataset directories to D: drive

### DISCUSSION THREAD

**Discussion by towanimodorenai** [MIC-DKFZ/nnUNet#2784] [Q&A]
Hi everyone,

I’m seeing persistent “Message for the error 5: Access is denied.” spam in the console whenever I run:

    nnunetv2_train MY_DATASET 2d 0 --npz

– It didn’t happen originally. The errors started appearing right after I moved my nnUNet_raw, nnUNet_preprocessed, and nnUNet_results folders from C: to D: and adjusted relevant environment variables (e.g. nnUNet_raw, nnUNet_preprocessed, nnUNet_results).
– I then moved everything back to C: and reset the env vars, but the “Access is denied” messages still persist.

I’ve already tried:

    - Running in an elevated (Administrator) shell

    - Setting nnUNet_n_proc_DA=0 to disable augmentation workers

    - Disabling memory‐mapped I/O by removing any mmap_mode options in np.load calls

    - Reinstalling the latest GitHub source in editable mode

Nothing has stopped the “Access is denied” messages from appearing at the start of each epoch.

Environment:

Windows 10
Python 3.13, Conda environment
nnU-Net v2.6.0 (installed from GitHub)
CUDA 11.8, single GPU (A6000)

Has anyone encountered this on Windows and found a reliable workaround? Thanks in advance!

**Comment by towanimodorenai** [comment]
(ct) C:\Windows\system32>nnunetv2_train 777 2d 0

############################
INFO: You are using the old nnU-Net default plans. We have updated our recommendations. Please consider using those instead! Read more here: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/resenc_presets.md
############################

Using device: cuda:0

#######################################################################
Please cite the following paper when using nnU-Net:
Isensee, F., Jaeger, P. F., Kohl, S. A., Petersen, J., & Maier-Hein, K. H. (2021). nnU-Net: a self-configuring method for deep learning-based biomedical image segmentation. Nature methods, 18(2), 203-211.
#######################################################################

2025-04-16 15:55:32.842462: do_dummy_2d_data_aug: False
2025-04-16 15:55:32.844419: Using splits from existing split file: D:\testfile\nnUNet_preprocessed\Dataset777_Mammo\splits_final.json
2025-04-16 15:55:32.846551: The split file contains 5 splits.
2025-04-16 15:55:32.847682: Desired fold for training: 0
2025-04-16 15:55:32.856426: This split has 378 training and 95 validation cases.
Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.


This is the configuration used by this training:
Configuration name: 2d
 {'data_identifier': 'nnUNetPlans_2d', 'preprocessor_name': 'DefaultPreprocessor', 'batch_size': 2, 'patch_size': [1536, 1024], 'median_image_size_in_voxels': [5796.0, 4726.0], 'spacing': [1.0, 1.0], 'normalization_schemes': ['ZScoreNormalization'], 'use_mask_for_norm': [False], 'resampling_fn_data': 'resample_data_or_seg_to_shape', 'resampling_fn_seg': 'resample_data_or_seg_to_shape', 'resampling_fn_data_kwargs': {'is_seg': False, 'order': 3, 'order_z': 0, 'force_separate_z': None}, 'resampling_fn_seg_kwargs': {'is_seg': True, 'order': 1, 'order_z': 0, 'force_separate_z': None}, 'resampling_fn_probabilities': 'resample_data_or_seg_to_shape', 'resampling_fn_probabilities_kwargs': {'is_seg': False, 'order': 1, 'order_z': 0, 'force_separate_z': None}, 'architecture': {'network_class_name': 'dynamic_network_architectures.architectures.unet.PlainConvUNet', 'arch_kwargs': {'n_stages': 9, 'features_per_stage': [32, 64, 128, 256, 512, 512, 512, 512, 512], 'conv_op': 'torch.nn.modules.conv.Conv2d', 'kernel_sizes': [[3, 3], [3, 3], [3, 3], [3, 3], [3, 3], [3, 3], [3, 3], [3, 3], [3, 3]], 'strides': [[1, 1], [2, 2], [2, 2], [2, 2], [2, 2], [2, 2], [2, 2], [2, 2], [2, 2]], 'n_conv_per_stage': [2, 2, 2, 2, 2, 2, 2, 2, 2], 'n_conv_per_stage_decoder': [2, 2, 2, 2, 2, 2, 2, 2], 'conv_bias': True, 'norm_op': 'torch.nn.modules.instancenorm.InstanceNorm2d', 'norm_op_kwargs': {'eps': 1e-05, 'affine': True}, 'dropout_op': None, 'dropout_op_kwargs': None, 'nonlin': 'torch.nn.LeakyReLU', 'nonlin_kwargs': {'inplace': True}}, '_kw_requires_import': ['conv_op', 'norm_op', 'dropout_op', 'nonlin']}, 'batch_dice': True}

These are the global plan.json settings:
 {'dataset_name': 'Dataset777_Mammo', 'plans_name': 'nnUNetPlans', 'original_median_spacing_after_transp': [999.0, 1.0, 1.0], 'original_median_shape_after_transp': [1, 5796, 4726], 'image_reader_writer': 'NaturalImage2DIO', 'transpose_forward': [0, 1, 2], 'transpose_backward': [0, 1, 2], 'experiment_planner_used': 'ExperimentPlanner', 'label_manager': 'LabelManager', 'foreground_intensity_properties_per_channel': {'0': {'max': 255.0, 'mean': 155.59985664346516, 'median': 158.0, 'min': 0.0, 'percentile_00_5': 61.0, 'percentile_99_5': 238.0, 'std': 31.30065810391188}}}

2025-04-16 15:55:33.212921: Unable to plot network architecture:
2025-04-16 15:55:33.214494: No module named 'hiddenlayer'
2025-04-16 15:55:33.284200:
2025-04-16 15:55:33.285956: Epoch 0
2025-04-16 15:55:33.287634: Current learning rate: 0.01
Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.

Message for the error 5:
Access is denied.
